owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/settings/SettingsPictureUploadsViewModel.kt

```diff
@@ -0,0 +1,91 @@
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.presentation.viewmodels.settings
+
+import android.content.Intent
+import androidx.lifecycle.ViewModel
+import com.owncloud.android.data.preferences.datasources.SharedPreferencesProvider
+import com.owncloud.android.datamodel.OCFile
+import com.owncloud.android.db.PreferenceManager
+import com.owncloud.android.providers.CameraUploadsHandlerProvider
+import com.owncloud.android.ui.activity.LocalFolderPickerActivity
+import com.owncloud.android.ui.activity.UploadPathActivity
+import java.io.File
+
+class SettingsPictureUploadsViewModel(
+    private val preferencesProvider: SharedPreferencesProvider,
+    private val cameraUploadsHandlerProvider: CameraUploadsHandlerProvider
+) : ViewModel() {
+
+    private var uploadPath: String? = null
+    private var uploadSourcePath: String? = null
+
+    fun isPictureUploadEnabled() =
+        preferencesProvider.getBoolean(PreferenceManager.PREF__CAMERA_PICTURE_UPLOADS_ENABLED, false)
+
+    fun setEnablePictureUpload(value: Boolean) =
+        preferencesProvider.putBoolean(PreferenceManager.PREF__CAMERA_PICTURE_UPLOADS_ENABLED, value)
+
+    fun updatePicturesLastSync() = cameraUploadsHandlerProvider.updatePicturesLastSync(0)
+
+    fun loadPictureUploadsPath() {
+        uploadPath = preferencesProvider.getString(
+            PreferenceManager.PREF__CAMERA_PICTURE_UPLOADS_PATH,
+            PreferenceManager.PREF__CAMERA_UPLOADS_DEFAULT_PATH
+        )
+    }
+
+    fun getPictureUploadsPath() = uploadPath
+
+    fun loadPictureUploadsSourcePath() {
+        uploadSourcePath = preferencesProvider.getString(
+            PreferenceManager.PREF__CAMERA_PICTURE_UPLOADS_SOURCE,
+            PreferenceManager.CameraUploadsConfiguration.getDefaultSourcePath()
+        )
+    }
+
+    fun getPictureUploadsSourcePath() = uploadSourcePath
+
+    fun handleSelectPictureUploadsPath(data: Intent?) {
+        val folderToUpload = data?.getParcelableExtra<OCFile>(UploadPathActivity.EXTRA_FOLDER)
+        uploadPath = folderToUpload?.remotePath
+        preferencesProvider.putString(PreferenceManager.PREF__CAMERA_PICTURE_UPLOADS_PATH, uploadPath!!)
+    }
+
+    fun handleSelectPictureUploadsSourcePath(data: Intent?) {
+        // If the source path has changed, update camera uploads last sync
+        var previousSourcePath = uploadSourcePath
+
+        if (previousSourcePath?.endsWith(File.separator) == true) {
+            previousSourcePath = previousSourcePath.substring(0, previousSourcePath.length - 1)
+        }
```

Comment:
What about `previousSourcePath?.trimEnd(File.separatorChar)`? Does it do the job?